src/components/CategoryFilter.js

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import categories from '../data/categories';
+
+const CategoryFilter = props => {
+  const { handleFilterChange, filter } = props;
+
+  useEffect(() => {
+    const select = document.querySelector('.filter');
+    if (filter === 'All') {
+      select.selectedIndex = 0;
+    } else {
+      select.selectedIndex = categories.indexOf(filter) + 1;
+    }
+  }, [filter]);
+
+  return (
+    <select
+      className="filter"
+      name="categoryFilter"
+      onChange={e => handleFilterChange(e.target.value)}
+    >
+      <option value="All">All</option>
+      {categories.map(cat => (
+        <option key={Math.floor(Math.random() * 999)} value={cat}>
+          {cat}
```

Comment:
**[Optional]** We can use the **cat** as a key for each option. It will eliminate 2 operations for each option in each render cycle.